### PR TITLE
feat: Refactor Weekly Summary Report

### PR DESCRIPTION
**Pull request by JoseAngel1196**
## This Change

In this PR, besides refactoring the weekly summary report, we're doing some refactoring across the repo, such as:

* Renaming the `config` module to `configs. There was a circular import error going on because the .go files inside that package were not extending from the config package which is why some of the functions were not found.
* Ensuring that if we don't find an icon, we return the default instead of returning an error. This will simplify the codebase.
* Creating a `test` package so we can reuse the test data across the repo.

And lastly, refactoring the weekly summary report and this is what it will look at the end ([link](https://app.slack.com/block-kit-builder/TCDBUV79T#%7B%22blocks%22:%5B%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:robot_face:%20%20Vulnbot%20Summary%20Report%20%20:robot_face:%22%7D%7D,%7B%22type%22:%22context%22,%22elements%22:%5B%7B%22text%22:%22_February%2012,%202024_%22,%22type%22:%22mrkdwn%22%7D%5D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Total%20Vulnerabilities*:%2038%5Cn*Affected%20Repositories*:%2020%5Cn*Total%20Repositories*:%20130%22%7D%5D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Team%20Breakdown%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:first_place_medal:%20*Team%20Apex*:%203%20vulnerabilities%22%7D,%22accessory%22:%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:red_circle:%201%20Critical%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_orange_circle:%201%20High%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_yellow_circle:%201%20Moderate%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_blue_circle:%201%20Low%22,%22emoji%22:true%7D%7D%5D%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:second_place_medal:%20*Team%20Phoenix*:%2016%20vulnerabilities%22%7D,%22accessory%22:%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:red_circle:%201%20Critical%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_orange_circle:%201%20High%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_yellow_circle:%201%20Moderate%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_blue_circle:%201%20Low%22,%22emoji%22:true%7D%7D%5D%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:third_place_medal:%20*Team%20Horizon*:%2025%20vulnerabilities%22%7D,%22accessory%22:%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:red_circle:%201%20Critical%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_orange_circle:%201%20High%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_yellow_circle:%201%20Moderate%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_blue_circle:%201%20Low%22,%22emoji%22:true%7D%7D%5D%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:small_blue_diamond:%20*Team%20Batman*:%2030%20vulnerabilities%22%7D,%22accessory%22:%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:red_circle:%201%20Critical%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_orange_circle:%201%20High%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_yellow_circle:%201%20Moderate%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_blue_circle:%201%20Low%22,%22emoji%22:true%7D%7D%5D%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:small_blue_diamond:%20*Team%20Nebula*:%2035%20vulnerabilities%22%7D,%22accessory%22:%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:red_circle:%201%20Critical%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_orange_circle:%201%20High%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_yellow_circle:%201%20Moderate%22,%22emoji%22:true%7D%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:large_blue_circle:%201%20Low%22,%22emoji%22:true%7D%7D%5D%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Severity%20Breakdown%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:red_circle:%20%20*Critical*:%203%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:large_orange_circle:%20%20*High*:%208%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:large_yellow_circle:%20%20*Moderate*:%2010%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:large_blue_circle:%20%20*Low*:%2030%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Ecosystem%20Breakdown%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:python:%20%20*Python*:%2010%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:javascript:%20%20*Javascript*:%208%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22:runner:%20%20*Go*:%202%22%7D%7D%5D%7D)):

Screenshot 2024-02-20 at 2 04 57 PM

### CODE REVIEW

File: configs/config.go [hunk 1]
```diff
@@ -89,22 +91,23 @@ func GetUserConfig(configFile string) (Config, error) {
...
 	for _, config := range severities {
 		if config.Label == SeverityNames[severity] {
-			return config.Slack_emoji, nil
+			return config.Slack_emoji
 		}
 	}
-	return "", fmt.Errorf("No Slack icon available for severity %s", SeverityNames[severity])
+	return DEFAULT_SLACK_ICON
+
 }
```
Comment (JoseAngel1196) [inline configs/config.go:102]: Instead of returning an error if we don't find an icon, i think it's better if we fallback to the default icon. Should simplify the codebase.

File: reporting/console.go [hunk 2]
```diff
@@ -24,27 +25,28 @@ func (c *ConsoleReporter) SendSummaryReport(
 	numRepos int,
 	report FindingSummary,
 	reportTime time.Time,
+	teamSummaries TeamSummaries,
 	wg *sync.WaitGroup,
 ) error {
 	defer wg.Done()
 	summaryReport := color.Bold.Sprint(header) + "\n"
-	summaryReport += color.Style{color.OpItalic}.Sprint(reportTime.Format(time.RFC1123)) + "\n\n"
+	summaryReport += color.Style{color.OpItalic}.Sprint(reportTime.Format(DATE_LAYOUT)) + "\n\n"
```
Comment (developerDemetri) [inline reporting/console.go:33]: [q] why the change away from a standard RFC format?
Do we just want to cut off all the time + tz info?

File: reporting/slack.go [hunk 3]
```diff
@@ -35,77 +36,45 @@ func (s *SlackReporter) BuildSummaryReport(
...
+	teamSummaries TeamSummaries,
 ) slack.Message {
 	reportBlocks := []slack.Block{
 		slack.NewHeaderBlock(
 			slack.NewTextBlockObject(slack.PlainTextType, header, false, false),
 		),
-		slack.NewDividerBlock(),
 		slack.NewContextBlock("", slack.NewTextBlockObject(
-			slack.PlainTextType, reportTime.Format(time.RFC1123), false, false,
+			slack.PlainTextType, reportTime.Format(DATE_LAYOUT), false, false,
```
Comment (developerDemetri) [inline reporting/slack.go:46]: [q] same as above, are we moving away from standard RFC formats to omit time info?

**Review by developerDemetri — APPROVED** [review]
Looks great, just 1 question around changing up the timestamp formats

File: reporting/console.go (hunk 2 above)
Comment (JoseAngel1196) [reply reporting/console.go:33]: yes, it was intentional haha, steve suggested that here https://getpinwheel.slack.com/archives/C01J0UC6MK8/p1708461855360679?thread_ts=1708375594.246659&cid=C01J0UC6MK8
